### DISCUSSION TITLE
Bun bundle html and typescript

### DISCUSSION THREAD

**Discussion by Duffscs** [oven-sh/bun#5358] [Q&A] [answered]
It's a basic parcel use case, can bun do this?
I can't find any info on this in the bundler and watch mode docs.

```index.html
<!-- index.html -->
<html>
  <body>
    <script src="./index.ts"></script>
  </body>
</html>
```

```js
// index.ts
import message from './message'
console.log(message)
```

```js
// message.ts
export default 'Hello, world'
```

**Comment by bakedPotatoLord** [comment]
I have the same question. I'm trying to do the same thing, with a file structure that looks like this:

![image](https://github.com/oven-sh/bun/assets/93680829/3ac75e17-932c-491a-8a44-36b95faa6993)

and when I run 

```ts
Bun.build({
  entrypoints: ['./demo/minimaldemo.html'],
  outdir: './builtDemo',
  naming: {
    // default values
    entry: '[dir]/[name].[ext]',
    chunk: '[name]-[hash].[ext]',
    asset: '[dir]/[name]-[hash][ext]',
  },
})
```

It acts weird, deciding that the `.html` file is an asset, and then turning the `.ts` into

```js
// demo/minimaldemo.html
var minimaldemo_default = "./minimaldemo-a02eb42fdf5d8158html";
export {
  minimaldemo_default as default
};
```

I'd expect it to transform the TypeScript into JavaScript and then change the file extension in the html.

**Reply by canadaduane** [reply]
I'm still looking for the answer to this question. The situation is precisely as @bakedPotatoLord says on the latest stable bun, `1.1.43`.

**Reply by kravetsone** [reply, marked as answer]
> I'm still looking for the answer to this question. The situation is precisely as @bakedPotatoLord says on the latest stable bun, `1.1.43`.

Use --experimental-html

**Reply by canadaduane** [reply]
I'm using Bun.build, rather than the CLI, and although it's hard to tell, I think the option is supposed to be `experimentalHtml: true` in the build options. However, even this does not seem to work (although it doesn't break the build, which an incorrect option seems to do).
